qiskit_dynamics/signals/signals.py

```diff
@@ -124,43 +123,44 @@ def is_constant(self) -> bool:
         return self._is_constant
 
     @property
-    def carrier_freq(self) -> Array:
+    def carrier_freq(self):
+        # TODO: specify a return type
         """The carrier frequency of the signal."""
         return self._carrier_freq
 
     @carrier_freq.setter
-    def carrier_freq(self, carrier_freq: Union[float, list, Array]):
+    def carrier_freq(self, carrier_freq: Union[float, list]):
         """Carrier frequency setter. List handling is to support subclasses storing a
         list of frequencies."""
-        if type(carrier_freq) == list:
-            carrier_freq = [Array(entry).data for entry in carrier_freq]
-        self._carrier_freq = Array(carrier_freq)
+        self._carrier_freq = unp.asarray(carrier_freq)
         self._carrier_arg = 1j * 2 * np.pi * self._carrier_freq
 
     @property
-    def phase(self) -> Array:
+    def phase(self):
+        # TODO: specify a return type
         """The phase of the signal."""
         return self._phase
 
     @phase.setter
-    def phase(self, phase: Union[float, list, Array]):
+    def phase(self, phase: Union[float, list]):
         """Phase setter. List handling is to support subclasses storing a
         list of phases."""
-        if type(phase) == list:
-            phase = [Array(entry).data for entry in phase]
-        self._phase = Array(phase)
+        self._phase = unp.asarray(phase)
```

Comment:
It is one approach of restricting to only numpy.array or jax.numpy.array in type of attributes in dynamics.